New block in the Symfony debug toolbar to switch the view

JonasHaouzi opened the pull request:
Hello guys.

I added a new Symfony toolbar (through the data collector) to switch easily to any view.

![screen shot 2015-08-20 at 10 12 42](https://cloud.githubusercontent.com/assets/8245107/9378730/07871046-4724-11e5-9cb3-12fb9462742e.png)

The update is really light but useful when debugging and working in project.

raulferras commented:
+1 That's a nice addition

xabbuh commented on DataCollector/DeviceDataCollector.php:
```diff
@@ -0,0 +1,112 @@
+    ) {
+        $this->data['currentView'] = $this->deviceView->getViewType();
+        $this->data['views'] = [
+            [
```
MobileDetectBundle claims to be compatible with PHP 5.3. Thus, you have to avoid the short array syntax.

xabbuh commented on DependencyInjection/MobileDetectExtension.php:
```diff
@@ -49,6 +49,7 @@ public function load(array $configs, ContainerBuilder $container)
 
         $container->setParameter('mobile_detect.mobile_detector.class', $config['mobile_detector_class']);
         $container->setParameter('mobile_detect.device_view.class', $config['device_view_class']);
+        $container->setParameter('mobile_detect.data_collector_device.class', $config['data_collector_device_class']);
```
I would not add a parameter for this (it's one extension point that strictly taken must be maintained). By the way, I don't see where you actually register the service as a data collector.

Quberik commented:
+1 nice feature

JonasHaouzi commented:
I fixed the problem with PHP 5.3 incompatibility, and made the updates related to your feedbacks.

I also fixed a problem when generating the URL to switch the device.

xabbuh commented on README.md:
```diff
@@ -102,6 +102,7 @@ mobile_detect:
     device_view_class: SunCat\MobileDetectBundle\Helper\DeviceView
     request_listener_class: SunCat\MobileDetectBundle\EventListener\RequestListener
     extension_class: SunCat\MobileDetectBundle\Twig\Extension\MobileDetectExtension
+    data_collector_device_class: SunCat/MobileDetectBundle/DataCollector/DeviceDataCollector
```
This option does not exist.
